sdk/search/Azure.Search.Documents/src/Generated/Models/Tokenizer.cs

```diff
@@ -14,6 +16,11 @@ public partial class Tokenizer
         /// <param name="name"> The name of the tokenizer. It must only contain letters, digits, spaces, dashes or underscores, can only start and end with alphanumeric characters, and is limited to 128 characters. </param>
         public Tokenizer(string name)
         {
+            if (name == null)
```

Comment:
1. Why not use our `Argument` validator class?
2. Maybe this is hard to codify because an empty string might be valid, but what about empty strings?
3. If you're using `== null` generally for reference types, you really should use `is null` which won't use an overridden `Equals`, i.e. it compiles to `object.ReferenceEquals`.

Reply:
Filed https://github.com/Azure/autorest.csharp/issues/577

I think checking for empty strings gets into a shady client validation territory and our guidance state that we shouldn't do excessive client validation and the service should throw.